### ISSUE TITLE
Exception thrown for undecorated optional boolean fields when using plugin

### ISSUE THREAD

**Issue by msheakoski** [nestjs/graphql#946]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
With the `@nestjs/graphql/plugin` enabled and given the file:

```typescript
// user-update.input.ts
@InputType()
export class UserUpdateInput {
  name?: string
  isEnabled?: boolean
}
```

an exception is thrown:

```
UnhandledPromiseRejectionWarning: Error: Cannot determine a GraphQL input type for the "isEnabled". Make sure your class is decorated with an appropriate decorator.
  at InputTypeFactory.create (/example/node_modules/@nestjs/graphql/dist/schema-builder/factories/input-type.factory.js:20:27)
  at /example/node_modules/@nestjs/graphql/dist/schema-builder/factories/input-type-definition.factory.js:45:56
  at Array.forEach (<anonymous>)
  at /example/node_modules/@nestjs/graphql/dist/schema-builder/factories/input-type-definition.factory.js:43:37
  at resolveThunk (/example/node_modules/graphql/type/definition.js:478:40)
  at defineInputFieldMap (/example/node_modules/graphql/type/definition.js:1203:18)
  at GraphQLInputObjectType.getFields (/example/node_modules/graphql/type/definition.js:1151:27)
  at collectReferencedTypes (/example/node_modules/graphql/type/schema.js:376:81)
  at collectReferencedTypes (/example/node_modules/graphql/type/schema.js:372:11)
  at new GraphQLSchema (/example/node_modules/graphql/type/schema.js:157:7)
```

## Expected behavior
The following schema should be generated:
```graphql
input UserUpdateInput {
  name: String
  isEnabled: Boolean
}
```

## Minimal reproduction of the problem with instructions
❌ Does not work
```typescript
// user-update.input.ts
@InputType()
export class UserUpdateInput {
  name?: string
  isEnabled?: boolean
}
```

✅  Works (when using `@Field` decorator)
```typescript
@InputType()
export class UserUpdateInput {
  name?: string

  @Field((type) => Boolean)
  isEnabled?: boolean
}
```

✅  Works (when using ! instead of ?)
```typescript
@InputType()
export class UserUpdateInput {
  name?: string
  isEnabled!: boolean
}
```

## Environment

<pre><code>
Nest version: 7.1.3

 
For Tooling issues:
- Node version: v14.4.0
- Platform: macOS 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by msheakoski** [comment]
@kamilmysliwiec Here is a link to the repository: https://github.com/msheakoski/nest-graphql-issue-946

The issue seems to happen only when `strict: true` is enabled in `tsconfig.json`. All relevant code is in the "example" module.

When switching between true/false, you must run `rm -rf dist` before running `npm run start:dev` because the last successful or failing compilation will be cached.

**Comment by trubit** [comment]
I just hit this error too
Problem is, that type text looks like `boolean | undefined` but TS returns actually union of 3 types `undefined | true | false`.
Thats when this line https://github.com/nestjs/graphql/blob/master/lib/plugin/utils/plugin-utils.ts#L79 fails. And code falls to `Object` type.

**Comment by kamilmysliwiec** [comment]
Fixed in 7.4.3
